controllers/certificaterequest/utils_test.go

```diff
@@ -0,0 +1,114 @@
+package certificaterequest
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type MockClient struct {
+	GetFunc func(ctx context.Context, key types.NamespacedName, obj client.Object, opts ...client.GetOption) error
+}
+
+func (m *MockClient) Get(ctx context.Context, key types.NamespacedName, obj client.Object, opts ...client.GetOption) error {
+	return m.GetFunc(ctx, key, obj, opts...)
+}
+
+func (m *MockClient) List(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error {
+	return nil
+}
+
+func (m *MockClient) Create(ctx context.Context, obj client.Object, opts ...client.CreateOption) error {
+	return nil
+}
+
+func (m *MockClient) Delete(ctx context.Context, obj client.Object, opts ...client.DeleteOption) error {
+	return nil
+}
+
+func (m *MockClient) Update(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
+	return nil
+}
+
+func (m *MockClient) Patch(ctx context.Context, obj client.Object, patch client.Patch, opts ...client.PatchOption) error {
+	return nil
+}
+
+func (m *MockClient) DeleteAllOf(ctx context.Context, obj client.Object, opts ...client.DeleteAllOfOption) error {
+	return nil
+}
+
+func (m *MockClient) Status() client.StatusWriter {
+	return nil
+}
+
+func (m *MockClient) Scheme() *runtime.Scheme {
+	return nil
+}
+
+func (m *MockClient) RESTMapper() meta.RESTMapper {
+	return nil
+}
+
+func (m *MockClient) SubResource(subResource string) client.SubResourceClient {
+	return nil
+}
```

Comment:
I believe controller runtime has a built-in mock client that we can utilize instead of building our own here.